dataland-internal-storage/src/main/kotlin/org/dataland/datalandinternalstorage/services/DatabaseDataStore.kt

```diff
@@ -1,28 +1,73 @@
 package org.dataland.datalandinternalstorage.services
 
+import com.fasterxml.jackson.databind.ObjectMapper
+import org.dataland.datalandbackend.openApiClient.api.NonPersistedDataControllerApi
 import org.dataland.datalandinternalstorage.entities.DataItem
 import org.dataland.datalandinternalstorage.repositories.DataItemRepository
+import org.dataland.datalandmessagequeueutils.cloudevents.CloudEventMessageHandler
+import org.slf4j.LoggerFactory
+import org.springframework.amqp.core.Message
+import org.springframework.amqp.rabbit.annotation.RabbitHandler
+import org.springframework.amqp.rabbit.annotation.RabbitListener
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
-import java.util.UUID
+import java.lang.IllegalArgumentException
 
 /**
  * Simple implementation of a data store using a postgres database
+ * @param dataItemRepository
+ * @param cloudEventMessageHandler service for managing CloudEvents messages
+ * @param nonPersistedDataClient the service for retrieving data from the temporary storage
+ * @param objectMapper object mapper used for converting data classes to strings and vice versa
  */
 @Component
 class DatabaseDataStore(
     @Autowired private var dataItemRepository: DataItemRepository,
-) {
+    @Autowired var cloudEventMessageHandler: CloudEventMessageHandler,
+    @Autowired var nonPersistedDataClient: NonPersistedDataControllerApi,
+    @Autowired var objectMapper: ObjectMapper,
+) { companion object {
+    private const val storageQueue = ("\${spring.rabbitmq.storage-queue}")
+    private const val storedQueue = ("\${spring.rabbitmq.stored-queue}")
+}
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Method that listens to the storage_queue and stores data into the database in case there is a message on the
+     * storage_queue
+     * @param message Message retrieved from storage_queue
+     */
+
+    @RabbitListener(queues = [storageQueue])
```

Comment:
Do we have any control over how these message are executed? In Sequence, In parallel? Number of parallell threads?

Reply:
https://docs.spring.io/spring-amqp/reference/html/#listener-concurrency gives the answer.
By default: One consumer, in sequence.